solarforecastarbiter/datamodel.py

```diff
@@ -780,8 +783,41 @@ class ProbabilisticForecast(
     def __post_init__(self):
         super().__post_init__()
         __check_axis__(self.axis)
+        __set_constant_values__(self)
         __check_axis_consistency__(self.axis, self.constant_values)
 
+    @classmethod
+    def from_dict(model, input_dict, raise_on_extra=False):
+        dict_ = input_dict.copy()
+        constant_values = dict_.pop('constant_values')
+        new_cvs = []
+        for cv in constant_values:
+            if isinstance(cv, dict):
+                new_cvs.append(
+                    ProbabilisticForecastConstantValue.from_dict(cv))
+            else:
+                new_cvs.append(cv)
+        dict_['constant_values'] = tuple(new_cvs)
+        return super().from_dict(dict_, raise_on_extra)
+
+
+def __set_constant_values__(self):
+    out = []
+    for cv in self.constant_values:
```

Comment:
looks like you could supply both `ProbabilisticForecastConstantValue` and `float` objects. I don't think we need to protect against that, just call it an implementation quirk.

Reply:
agreed

Reply:
agreed